gdpr/fields.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict, List, TYPE_CHECKING, Tuple, Type, Union
+
+from django.db.models import Model
+
+from gdpr.loading import anonymizer_register
+
+FieldMatrix = Union[str, Tuple[Any, ...]]
+FieldList = Union[List[str], str]
+RelatedFieldDict = Dict[str, "Fields"]
+
+if TYPE_CHECKING:
+    from gdpr.anonymizers import ModelAnonymizer
+
+
+class Fields:
+    local_fields: FieldList
+    related_fields: RelatedFieldDict
+    anonymizer: "ModelAnonymizer"
+    model: Type[Model]
+
+    def __init__(self, fields: FieldMatrix, model: Type[Model], anonymizer_instance: "ModelAnonymizer" = None):
+        self.model = model
+        self.anonymizer = anonymizer_register[self.model]() if anonymizer_instance is None else anonymizer_instance
+        self.local_fields = self.parse_local_fields(fields)
+        self.related_fields = self.parse_related_fields(fields)
+
+    def parse_local_fields(self, fields: FieldMatrix) -> FieldList:
+        """Get Iterable of local fields from fields matrix."""
+        if fields == "__ALL__" or "__ALL__" in fields:
```

Comment:
přoč by se stalo "__ALL__" in fields? To moc nedává smysl ne? Navíc kdyby někdo použil fields = '__ALL__cokoli' tak ti také projde true. BTW to ALL bych si dal do konstanty

Reply:
opraveno